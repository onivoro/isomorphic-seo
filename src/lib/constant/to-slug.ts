
export const toSlug = (s: string, delim?: string) => {
    const d = delim || '-';
    console.log(s, delim, d);
    // const singleSpace = ' ';
    // const delim = _delim || singleSpace;
    return s
        // .replace(/\s+/g, delim || singleSpace)
        // .replace(/,\s/g, delim || '')
        .replace(/\W+/g, d)
        .toLowerCase();
};
