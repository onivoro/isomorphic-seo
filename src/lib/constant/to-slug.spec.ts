import { toSlug } from './to-slug';

const testPhrase = 'test  phrase';
const testWord = 'test';

describe('toSlug', () => {
    describe('success case', () => {
        it.each([
            [testWord, null],
            [', ', null],
            [testPhrase, null],
            [testPhrase, '~'],
        ])('worx', (tp: string, delim: string) => {
            expect(toSlug(tp, delim))
                .toMatchSnapshot();
        });

    });
});
