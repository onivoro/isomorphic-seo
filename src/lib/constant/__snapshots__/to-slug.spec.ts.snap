// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toSlug success case worx 1`] = `"test"`;

exports[`toSlug success case worx 2`] = `"-"`;

exports[`toSlug success case worx 3`] = `"test-phrase"`;

exports[`toSlug success case worx 4`] = `"test~phrase"`;
